micropython/umqtt.simple/umqtt/simple.py

```diff
@@ -60,8 +60,10 @@ def set_last_will(self, topic, msg, retain=False, qos=0):
         self.lw_qos = qos
         self.lw_retain = retain
 
-    def connect(self, clean_session=True):
+    def connect(self, clean_session=True, timeout=None):
         self.sock = socket.socket()
+        if timeout is not None:
+            self.sock.settimeout(timeout)
```

Comment:
If you pass `None` to `settimeout` then it sets it to blocking mode, which is also the default after creating a fresh socket.

So, the `if timeout is not None` line is not needed.  It can just set the timeout unconditionally and still have the same behaviour as before.  Doing it that way makes the code smaller and simpler.

Reply:
Hi @dpgeorge ,
Agree with you. I submitted the change.
(I was trying to be little cautious by avoiding to execute settimeout when default value arrives! I agree, this is not a clean solution!)

Reply:
@dpgeorge 
May you please suggest the next step? 
Should I create a PR for this?

Reply:
This change looks good now, thank you.

But I see on your commits that the author is `Your Name <you@example.com>`.  Is that intentional?  Usually we require a proper name and email, for attribution of the work.

If you want to remain anonymous, I can change the author to my name.

Reply:
Hi @dpgeorge ,
Thank you for notifying me 🙏.
I have updated my Name/E-mail id and pushed it again.
Please let me know if I need to do something further.

Reply:
This looks good now, thank you.